packages/driver/src/cross-origin/patches/cookies.js

```diff
@@ -77,4 +136,30 @@ export const patchDocumentCookie = (Cypress, window) => {
   }
 
   window.addEventListener('unload', onUnload)
+
+  Cypress.on('window:before:load', patch)
+
+  Cypress.specBridgeCommunicator.on('cross:origin:cookies', (cookies) => {
+    updateCookies(cookies)
+
+    Cypress.state('crossOriginCookies', cookies)
+
+    Cypress.specBridgeCommunicator.toPrimary('cross:origin:cookies:received')
+  })
+
+  Cypress.on('test:before:run', () => {
+    reset()
+  })
+
+  Cypress.on('set:cookie', (cookie) => {
+    updateCookies([cookie])
+  })
+
+  Cypress.on('clear:cookie', (name) => {
+    documentCookieValue = clearCookie(documentCookieValue, name)
+  })
+
+  Cypress.on('clear:cookies', () => {
+    reset()
```

Comment:
I wonder if we still need to do this if we store the reference to `get:cookies`, or just clear the reference out?

Reply:
This is to optimistically clear out the cookies when a user calls `cy.clearCookies()` so there's no delay/race condition waiting to sync with the automation cookies. There's actually no longer any polling to sync with automation cookies. It just caused issues and now all cases of cookies changing in automation are taken care of in a more deterministic fashion